admin/query/src/main/java/org/codice/ditto/replication/admin/query/sites/persist/UpdateReplicationSite.java

```diff
@@ -71,12 +76,31 @@ public ReplicationSiteField getReturnType() {
   }
 
   @Override
-  public FunctionField<ReplicationSiteField> newInstance() {
+  public FunctionField<BooleanField> newInstance() {
     return new UpdateReplicationSite(replicationUtils);
   }
 
   @Override
   public Set<String> getFunctionErrorCodes() {
-    return ImmutableSet.of();
+    return ImmutableSet.of(ReplicationMessages.DUPLICATE_SITE, DefaultMessages.NO_EXISTING_CONFIG);
+  }
+
+  @Override
+  public void validate() {
+    super.validate();
+    if (containsErrorMsgs()) {
+      return;
+    }
+
+    final String updatedName = name.getValue();
+    final String updateId = id.getValue();
+    if (!replicationUtils.siteIdExists(updateId)) {
+      addErrorMessage(DefaultMessages.noExistingConfigError());
+    }
+    if (updatedName != null
+        && (replicationUtils.isNotUpdatedSiteName(updateId, updatedName)
+            || replicationUtils.isDuplicateSiteName(updatedName))) {
```

Comment:
If an update request is sent with the name of the site identified by the ID, the isDuplicateSiteName would not be sufficient because it would return true for its own name, but it should be allowed to update regardless if it is its own name. However this logic is wrong and I updated it :D